r/aws_appmesh_virtual_node: AWS App Mesh support for TLS in transit encryption

ewbankkit opened the pull request:
<!--- See what makes a good Pull Request at : https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md#pull-requests --->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

<!--- Thank you for keeping this note for the community --->

<!--- If your PR fully resolves and should automatically close the linked issue, use Closes. Otherwise, use Relates --->
Closes #10922.

Includes ACMPCA changes from #13684.
Replaces https://github.com/terraform-providers/terraform-provider-aws/pull/12528.
Cherry picked relevant commits from #11850.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):
<!--
If change is not user facing, just write "NONE" in the release-note block below.
-->

```release-note
resource/aws_appmesh_virtual_node: Add `spec.backend_defaults`, `spec.backend.virtual_service.client_policy` and `spec.listener.tls` attributes to support TLS in transit encryption
```

Output from acceptance testing:

<!--
Replace TestAccXXX with a pattern that matches the tests affected by this PR.

For more information on the `-run` flag, see the `go test` documentation at https://tip.golang.org/cmd/go/#hdr-Testing_flags.
-->
```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAppmesh'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppmesh -timeout 120m
=== RUN   TestAccAWSAppmesh
=== RUN   TestAccAWSAppmesh/Route
=== RUN   TestAccAWSAppmesh/Route/tags
=== RUN   TestAccAWSAppmesh/Route/httpHeader
=== RUN   TestAccAWSAppmesh/Route/httpRoute
=== RUN   TestAccAWSAppmesh/Route/tcpRoute
=== RUN   TestAccAWSAppmesh/Route/routePriority
=== RUN   TestAccAWSAppmesh/VirtualNode
=== RUN   TestAccAWSAppmesh/VirtualNode/logging
=== RUN   TestAccAWSAppmesh/VirtualNode/tls
=== RUN   TestAccAWSAppmesh/VirtualNode/tags
=== RUN   TestAccAWSAppmesh/VirtualNode/basic
=== RUN   TestAccAWSAppmesh/VirtualNode/backendDefaults
=== RUN   TestAccAWSAppmesh/VirtualNode/clientPolicyAcm
=== RUN   TestAccAWSAppmesh/VirtualNode/cloudMapServiceDiscovery
=== RUN   TestAccAWSAppmesh/VirtualNode/listenerHealthChecks
=== RUN   TestAccAWSAppmesh/VirtualNode/clientPolicyFile
=== RUN   TestAccAWSAppmesh/VirtualRouter
=== RUN   TestAccAWSAppmesh/VirtualRouter/basic
=== RUN   TestAccAWSAppmesh/VirtualRouter/tags
=== RUN   TestAccAWSAppmesh/VirtualService
=== RUN   TestAccAWSAppmesh/VirtualService/tags
=== RUN   TestAccAWSAppmesh/VirtualService/virtualNode
=== RUN   TestAccAWSAppmesh/VirtualService/virtualRouter
=== RUN   TestAccAWSAppmesh/Mesh
=== RUN   TestAccAWSAppmesh/Mesh/basic
=== RUN   TestAccAWSAppmesh/Mesh/egressFilter
=== RUN   TestAccAWSAppmesh/Mesh/tags
--- PASS: TestAccAWSAppmesh (1254.58s)
    --- PASS: TestAccAWSAppmesh/Route (279.11s)
        --- PASS: TestAccAWSAppmesh/Route/tags (73.22s)
        --- PASS: TestAccAWSAppmesh/Route/httpHeader (51.70s)
        --- PASS: TestAccAWSAppmesh/Route/httpRoute (51.45s)
        --- PASS: TestAccAWSAppmesh/Route/tcpRoute (51.61s)
        --- PASS: TestAccAWSAppmesh/Route/routePriority (51.13s)
    --- PASS: TestAccAWSAppmesh/VirtualNode (564.00s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/logging (46.39s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/tls (96.22s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/tags (65.99s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/basic (29.31s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/backendDefaults (46.03s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/clientPolicyAcm (69.99s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/cloudMapServiceDiscovery (117.82s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/listenerHealthChecks (46.01s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/clientPolicyFile (46.23s)
    --- PASS: TestAccAWSAppmesh/VirtualRouter (109.53s)
        --- PASS: TestAccAWSAppmesh/VirtualRouter/basic (45.67s)
        --- PASS: TestAccAWSAppmesh/VirtualRouter/tags (63.86s)
    --- PASS: TestAccAWSAppmesh/VirtualService (171.36s)
        --- PASS: TestAccAWSAppmesh/VirtualService/tags (72.34s)
        --- PASS: TestAccAWSAppmesh/VirtualService/virtualNode (51.69s)
        --- PASS: TestAccAWSAppmesh/VirtualService/virtualRouter (47.34s)
    --- PASS: TestAccAWSAppmesh/Mesh (130.58s)
        --- PASS: TestAccAWSAppmesh/Mesh/basic (24.52s)
        --- PASS: TestAccAWSAppmesh/Mesh/egressFilter (50.66s)
        --- PASS: TestAccAWSAppmesh/Mesh/tags (55.39s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1254.639s

$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAcmCertificate_privateCert'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAcmCertificate_privateCert -timeout 120m
=== RUN   TestAccAWSAcmCertificate_privateCert
=== PAUSE TestAccAWSAcmCertificate_privateCert
=== CONT  TestAccAWSAcmCertificate_privateCert
--- PASS: TestAccAWSAcmCertificate_privateCert (34.79s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	34.827s
```

ewbankkit commented on aws/resource_aws_acmpca_certificate_authority_test.go:
```diff
@@ -496,6 +520,85 @@ func testAccCheckAwsAcmpcaCertificateAuthorityExists(resourceName string, certif
 	}
 }
 
+func testAccCheckAwsAcmpcaCertificateAuthorityActivateCA(certificateAuthority *acmpca.CertificateAuthority) resource.TestCheckFunc {
```
This function activates a `ROOT` CA following the process described in the [documentation](https://docs.aws.amazon.com/acm-pca/latest/userguide/PCACertInstall.html#InstallRoot). It can be used in other tests where an `Active` CA is required.

ewbankkit commented on aws/resource_aws_acmpca_certificate_authority_test.go:
```diff
@@ -40,24 +42,45 @@ func testSweepAcmpcaCertificateAuthorities(region string) error {
 	for _, certificateAuthority := range certificateAuthorities {
 		arn := aws.StringValue(certificateAuthority.Arn)
+
+		if aws.StringValue(certificateAuthority.Status) == acmpca.CertificateAuthorityStatusActive {
```
CA can only be deleted if it's `Disabled`.

ewbankkit commented:
Waiting for account limit of number of CAs to be raised before I can continue testing.

ewbankkit commented:
The problem is caused by the ACM certificate being in `Failed` state when I try and assign it to the Virtual Node's TLS listener. This exposed a bug in the App Mesh service.
For the acceptance test we need to make sure that the associated PCA is activated before attempting to issue a cert.

ewbankkit commented:
```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAcmCertificate_privateCert'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAcmCertificate_privateCert -timeout 120m
=== RUN   TestAccAWSAcmCertificate_privateCert
=== PAUSE TestAccAWSAcmCertificate_privateCert
=== CONT  TestAccAWSAcmCertificate_privateCert
--- PASS: TestAccAWSAcmCertificate_privateCert (34.79s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	34.827s
```

ewbankkit commented:
Revisit, considering #13194.

ewbankkit commented:
#### TODO

* Remove `appmeshVirtualNodeBackendHash`
* Re-run and fix acceptance tests

mbamber commented:
Is this still a WIP? We'd like to get this merged in, so if there's something I can do to help let me know!

ewbankkit commented:
@mbamber Yes, still a _WIP_, mainly because of recent changes to the way some of the acceptance test checks are done; I need to fix some merge conflicts.

ewbankkit commented:
Remove _WIP_.

gurjant31 commented:
When this will be merged? I kind of need this feature

ewbankkit commented:
Rebased and fixed merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAppmesh/VirtualNode'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppmesh/VirtualNode -timeout 120m
=== RUN   TestAccAWSAppmesh_serial
=== RUN   TestAccAWSAppmesh_serial/VirtualNode
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/logging
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/tls
    resource_aws_appmesh_virtual_node_test.go:379: [INFO] Got non-empty plan, as expected
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/tags
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/backendDefaults
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/clientPolicyFile
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/basic
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/clientPolicyAcm
    resource_aws_appmesh_virtual_node_test.go:579: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSAppmesh_serial (457.60s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualNode (457.60s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks (34.63s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/logging (33.59s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tls (83.48s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tags (50.07s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/backendDefaults (33.62s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/clientPolicyFile (33.98s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery (106.66s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/basic (19.86s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/clientPolicyAcm (61.70s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	457.636s
```

rmaskara commented:
can we please get this reviewed and merged please. We would like to start testing with this change.

ewbankkit commented:
Rebased.
Acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAppmesh'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppmesh -timeout 120m
=== RUN   TestAccAWSAppmesh_serial
=== RUN   TestAccAWSAppmesh_serial/Route
=== RUN   TestAccAWSAppmesh_serial/Route/tcpRoute
=== RUN   TestAccAWSAppmesh_serial/Route/routePriority
=== RUN   TestAccAWSAppmesh_serial/Route/tags
=== RUN   TestAccAWSAppmesh_serial/Route/httpHeader
=== RUN   TestAccAWSAppmesh_serial/Route/httpRetryPolicy
=== RUN   TestAccAWSAppmesh_serial/Route/httpRoute
=== RUN   TestAccAWSAppmesh_serial/VirtualNode
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/tags
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/basic
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/backendDefaults
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/logging
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/tls
    resource_aws_appmesh_virtual_node_test.go:389: [INFO] Got non-empty plan, as expected
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/clientPolicyAcm
    resource_aws_appmesh_virtual_node_test.go:597: [INFO] Got non-empty plan, as expected
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/clientPolicyFile
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter/basic
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter/tags
=== RUN   TestAccAWSAppmesh_serial/VirtualService
=== RUN   TestAccAWSAppmesh_serial/VirtualService/virtualNode
=== RUN   TestAccAWSAppmesh_serial/VirtualService/virtualRouter
=== RUN   TestAccAWSAppmesh_serial/VirtualService/tags
=== RUN   TestAccAWSAppmesh_serial/Mesh
=== RUN   TestAccAWSAppmesh_serial/Mesh/tags
=== RUN   TestAccAWSAppmesh_serial/Mesh/basic
=== RUN   TestAccAWSAppmesh_serial/Mesh/egressFilter
--- PASS: TestAccAWSAppmesh_serial (1043.85s)
    --- PASS: TestAccAWSAppmesh_serial/Route (283.29s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tcpRoute (55.89s)
        --- PASS: TestAccAWSAppmesh_serial/Route/routePriority (38.56s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tags (56.34s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpHeader (39.17s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRetryPolicy (38.42s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRoute (54.90s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualNode (457.18s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tags (49.27s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/basic (19.13s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/backendDefaults (34.97s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery (106.01s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/logging (34.19s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tls (82.78s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/clientPolicyAcm (61.81s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/clientPolicyFile (34.46s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks (34.57s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualRouter (81.36s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/basic (33.13s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/tags (48.23s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualService (126.87s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualNode (37.93s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualRouter (34.42s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/tags (54.51s)
    --- PASS: TestAccAWSAppmesh_serial/Mesh (95.15s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/tags (41.77s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/basic (16.30s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/egressFilter (37.08s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1043.903s
```

DrFaust92 reviewed (approved):
LGTM

```
--- PASS: TestAccAWSAppmesh_serial (2435.37s)
    --- PASS: TestAccAWSAppmesh_serial/Mesh (284.03s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/egressFilter (111.26s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/tags (118.81s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/basic (53.95s)
    --- PASS: TestAccAWSAppmesh_serial/Route (703.35s)
        --- PASS: TestAccAWSAppmesh_serial/Route/routePriority (98.51s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tags (145.75s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpHeader (95.11s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRetryPolicy (90.76s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRoute (146.93s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tcpRoute (126.28s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualNode (933.34s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/clientPolicyFile (81.10s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tags (113.79s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/backendDefaults (71.51s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/clientPolicyAcm (117.58s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks (71.25s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/logging (81.20s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tls (190.93s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/basic (41.22s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery (164.75s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualRouter (182.88s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/basic (76.06s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/tags (106.83s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualService (331.76s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualNode (87.43s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualRouter (90.64s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/tags (153.69s)
```

breathingdust commented:
LGTM! 🚀  Thanks @ewbankkit 

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSAppmesh'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppmesh -timeout 120m
=== RUN   TestAccAWSAppmesh_serial
=== RUN   TestAccAWSAppmesh_serial/Mesh
=== RUN   TestAccAWSAppmesh_serial/Mesh/basic
=== RUN   TestAccAWSAppmesh_serial/Mesh/egressFilter
=== RUN   TestAccAWSAppmesh_serial/Mesh/tags
=== RUN   TestAccAWSAppmesh_serial/Route
=== RUN   TestAccAWSAppmesh_serial/Route/httpRetryPolicy
=== RUN   TestAccAWSAppmesh_serial/Route/httpRoute
=== RUN   TestAccAWSAppmesh_serial/Route/tcpRoute
=== RUN   TestAccAWSAppmesh_serial/Route/routePriority
=== RUN   TestAccAWSAppmesh_serial/Route/tags
=== RUN   TestAccAWSAppmesh_serial/Route/httpHeader
=== RUN   TestAccAWSAppmesh_serial/VirtualNode
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/clientPolicyFile
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/tags
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/basic
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/backendDefaults
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/logging
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/tls
    resource_aws_appmesh_virtual_node_test.go:389: [INFO] Got non-empty plan, as expected
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/clientPolicyAcm
    resource_aws_appmesh_virtual_node_test.go:597: [INFO] Got non-empty plan, as expected
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter/basic
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter/tags
=== RUN   TestAccAWSAppmesh_serial/VirtualService
=== RUN   TestAccAWSAppmesh_serial/VirtualService/virtualNode
=== RUN   TestAccAWSAppmesh_serial/VirtualService/virtualRouter
=== RUN   TestAccAWSAppmesh_serial/VirtualService/tags
--- PASS: TestAccAWSAppmesh_serial (887.25s)
    --- PASS: TestAccAWSAppmesh_serial/Mesh (87.56s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/basic (15.74s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/egressFilter (33.55s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/tags (38.27s)
    --- PASS: TestAccAWSAppmesh_serial/Route (220.31s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRetryPolicy (29.57s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRoute (44.76s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tcpRoute (43.11s)
        --- PASS: TestAccAWSAppmesh_serial/Route/routePriority (31.22s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tags (42.15s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpHeader (29.49s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualNode (405.14s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/clientPolicyFile (28.63s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery (116.60s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tags (40.09s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/basic (15.75s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/backendDefaults (29.97s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/logging (28.10s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tls (67.75s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/clientPolicyAcm (50.92s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks (27.32s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualRouter (68.96s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/basic (27.83s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/tags (41.13s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualService (105.28s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualNode (31.66s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualRouter (29.56s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/tags (44.06s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	889.109s

```
